migrations/legacy_intersection_type.go

```diff
@@ -43,6 +43,11 @@ func (t *LegacyIntersectionType) ID() common.TypeID {
 	}
 
 	var result strings.Builder
+
+	if t.LegacyType != nil {
+		result.WriteString(string(t.LegacyType.ID()))
+	}
```

Comment:
This was a bug in intersection-type hashing. The old restricted-type also includes the 'restricting' type in hashing.